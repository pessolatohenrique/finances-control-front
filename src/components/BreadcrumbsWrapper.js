import React from "react";
import { Typography, Link, Breadcrumbs } from "@mui/material";

function BreadcrumbsWrapper({ childrenLabel, parentLink }) {
  return (
    <Breadcrumbs separator="›" aria-label="breadcrumb">
      <Link
        underline="hover"
        key="1"
        color="inherit"
        href="/"
        variant="body2"
        data-testid="bread-firstlevel"
      >
        Início
      </Link>
      {parentLink && (
        <Link
          underline="hover"
          key="2"
          color="inherit"
          href={parentLink.link}
          variant="body2"
          data-testid="bread-secondlevel"
        >
          {parentLink.label}
        </Link>
      )}
      <Typography
        key="3"
        color="text.primary"
        variant="body2"
        data-testid="bread-lastlevel"
      >
        {childrenLabel}
      </Typography>
    </Breadcrumbs>
  );
}

export default BreadcrumbsWrapper;
