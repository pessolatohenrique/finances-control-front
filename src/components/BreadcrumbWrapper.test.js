import { render, screen } from "@testing-library/react";
import BreadcrumbsWrapper from "./BreadcrumbsWrapper";

describe("BreadcrumbWrapper", () => {
  it("should generate snapshot", () => {
    const { container } = render(
      <BreadcrumbsWrapper childrenLabel="Autores" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render when there is a level", () => {
    render(<BreadcrumbsWrapper childrenLabel="Autores" />);

    expect(screen.getByTestId("bread-firstlevel")).toHaveTextContent("Início");
    expect(screen.getByTestId("bread-lastlevel")).toHaveTextContent("Autores");
  });

  it("should render when there is two levels", () => {
    render(
      <BreadcrumbsWrapper
        parentLink={{ link: "autores", label: "Autores" }}
        childrenLabel="Novo"
      />
    );

    expect(screen.getByTestId("bread-firstlevel")).toHaveTextContent("Início");
    expect(screen.getByTestId("bread-secondlevel")).toHaveTextContent(
      "Autores"
    );
    expect(screen.getByTestId("bread-lastlevel")).toHaveTextContent("Novo");
  });
});
