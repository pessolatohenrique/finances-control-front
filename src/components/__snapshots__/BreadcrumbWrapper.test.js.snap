// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbWrapper should generate snapshot 1`] = `
<nav
  aria-label="breadcrumb"
  class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
>
  <ol
    class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
  >
    <li
      class="MuiBreadcrumbs-li"
    >
      <a
        class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineHover css-9smqfy-MuiTypography-root-MuiLink-root"
        data-testid="bread-firstlevel"
        href="/"
      >
        Início
      </a>
    </li>
    <li
      aria-hidden="true"
      class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
    >
      ›
    </li>
    <li
      class="MuiBreadcrumbs-li"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
        data-testid="bread-lastlevel"
      >
        Autores
      </p>
    </li>
  </ol>
</nav>
`;
